Ability to morph Profile by changing strike point coordinates only

mikekryjak opened the pull request:
Adds a new Profile method: the user specifies control point positions along the Profile as well as strike point coordinates. The method will return offsets that allow the Profile to be morphed accordingly. Great if you want a parameter scan of strike point locations.

![image](https://github.com/user-attachments/assets/95d2f62c-9c10-45f0-b1dd-0773348f6f5d)


- [x] Implement and test
- [x] Possibly refactor to not use Pandas since it's unnecessary (I did this because it was easier to debug when looking at a displayed dataframe)
- [x] Add to examples/tests

@LiamPattinson please ignore this one for our refactor works, don't let it hold you up. It can be merged after https://github.com/FusionDLS/FusionDLS/pull/39.

mikekryjak committed:
Add Profile method to allow morphing by specifying strike point position
The user specifies control point positions along the Profile as well as strike point coordinates. The method will return offsets that allow the Profile to be morphed accordingly. Great if you want a parameter scan of strike point locations.

mikekryjak committed:
Add tip that "pos" should be defined from X-point

mikekryjak committed:
Minor Profile plotting fixes

mikekryjak committed:
Minor fix in strike point offsets

mikekryjak committed:
Minor fixes in Profile.plot()
Spar_Spol had the axes the wrong way around, and mode="RZ" was missing the X-point.

LiamPattinson commented on src/fusiondls/Profile.py:
```diff
@@ -357,8 +361,68 @@ def recalculate_topology(
+
+        spl = cord_spline(self.R_leg, self.Z_leg, return_spline=True)
+
+        for i, pos in enumerate(cp["pos"]):
```
The linter is complaining here that the function argument `pos` is being overwritten in the for loop. Perhaps the function argument could just be renamed `position` and everything else could stay the same?

mikekryjak replied:
I don't get any warnings from Ruff on this at the moment. Let me know if it still shows up on your end because that means I have something misconfigured.

mikekryjak committed:
Merge branch 'main' into strike-point-offsets

mikekryjak committed:
Refactor get_offsets_strike_point to not need Pandas
That was unnecessary, Pandas is not used anywhere else in the class.

mikekryjak committed:
Add examples/tests for strike point interpolation

mikekryjak committed:
Remove unnecessary Pandas import in Profile

mikekryjak committed:
Formatting

mikekryjak committed:
Merge branch 'bugfix/make_magneticgeometry_from_profile' into strike-point-offsets

mikekryjak committed:
Merge branch 'tweak-settings' into strike-point-offsets

mikekryjak committed:
More formatting

mikekryjak committed:
Merge branch 'main' into strike-point-offsets

mikekryjak committed:
Merge branch 'add-postprocessing' into strike-point-offsets

mikekryjak committed:
Merge branch 'add-postprocessing' into strike-point-offsets

mikekryjak committed:
Merge branch 'add-postprocessing' into strike-point-offsets

mikekryjak committed:
Merge branch 'main' into strike-point-offsets

mikekryjak committed:
Improve get_offsets_strike_point docstrings/inputs

mikekryjak committed:
Change default morph to keep Bpol=const

Previously it was Bpitch=const, which avoids changing detachment sensitivity but turned out to create unrealistic flux compression for short divertor legs due to low Btor and high Bpol

mikekryjak commented:
@LiamPattinson this one is ready.
